pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
```

Comment:
> @deads2k I moved this field one level up to prevent from duplication in all the details structs.

Needs to be a pointer if its up here.  The server won't be able to validate every kind of signature.

I think I actually want it on the details, so its obvious which part of the information we've validated.